core/tests/puppeteer-acceptance-tests/blog-admin-tests/create-blog-post-draft.js

```diff
@@ -0,0 +1,22 @@
+const e2eBlogPostAdmin = require("../utility-functions/blogPostAdminUtils.js");
+const testConstants = require("../utility-functions/testConstants.js");
+
+const blogDashboardUrl = testConstants.URLs.BlogDashboard;
+const homePage = testConstants.Dashboard.LearnerDashboard;
+
+async function createDraftAsBlogPostAdmin() {
+  const blogPostAdmin = await new e2eBlogPostAdmin();
+  await blogPostAdmin.openBrowser();
+
+  await blogPostAdmin.signInWithEmail("testadmin@example.com");
+  await blogPostAdmin.waitForPageToLoad(homePage);
+  await blogPostAdmin.goto(blogDashboardUrl);
+  await blogPostAdmin.expectDraftBlogPostWithTitleToNotExist("Test Blog Post");
+  await blogPostAdmin.expectNumberOfDraftOrPublishedBlogPostsGreaterThan(0);
```

Comment:
Shouldn't this be "equal to zero"?

If the test is passing in this state then something may be wrong with your code.

Reply:
it depends.. my current implementation is if we are having draft blog post greater than 0, then we need to click '+New blog' for creating a new draft/publish blog post. if we expect there must be 0 blog posts as drafts/published then, we need to click 'CREATE NEW BLOG POST'. So this expectation affect some code of our internal implementation.

> Shouldn't this be "equal to zero"?
> 
> If the test is passing in this state then something may be wrong with your code.

Reply:
But just reading the test as-is, you haven't done anything yet on the dashboard, so why would the number of blog posts be anything other than 0?

Reply:
umm right.. so should I apply this change to all tests for blog-post

Reply:
I'm not sure what you mean. What change exactly are you planning to apply?

Basically, the requirement is that the method does the correct thing. If I am reading this correctly, you are saying that you expect the number of draft or published blog posts to be greater than 0, even though nothing has been set up on the blog dashboard yet. Isn't that incorrect?

Also, just to make sure we're on the same page -- the "expect" methods should just check that the condition holds and fail the test if not. They shouldn't be "doing" anything actively.

Reply:
yes our expect functions are just checking the condition, not doing any work
also, I think the expectation here must be, expectNumberOfDraftOrPublishedBlogPostsEqualTo(0)

Reply:
Yup I agree. But if this function is checking the wrong thing, then why are the tests passing?